libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialog.tsx

```diff
@@ -4,42 +4,69 @@ import { ShareDialogBase } from "./ShareDialogBase/ShareDialogBase";
 import { GranteeItem } from "./ShareDialogBase/types";
 import { IShareDialogProps } from "./types";
 import {
+    mapGranteesToAccessGrantees,
     mapGranteesToShareStatus,
     mapOwnerToGrantee,
-    mapShareStatusToGroupAll,
-    mapUserToInactiveGrantee,
+    mapUserToInactiveOwner,
 } from "./shareDialogMappers";
+import {
+    BackendProvider,
+    IntlWrapper,
+    UnexpectedSdkError,
+    useBackendStrict,
+    useWorkspaceStrict,
+    WorkspaceProvider,
+} from "@gooddata/sdk-ui";
 
 /**
  * @internal
  */
 export const ShareDialog: React.FC<IShareDialogProps> = (props) => {
-    const { sharedObject, currentUserRef, onApply, onCancel } = props;
-    const { createdBy, shareStatus } = sharedObject;
+    const { backend, workspace, locale, sharedObject, currentUserRef, onApply, onCancel, onError } = props;
+    const { createdBy, shareStatus, ref: sharedObjectRef } = sharedObject;
+
+    const effectiveBackend = useBackendStrict(backend);
+    const effectiveWorkspace = useWorkspaceStrict(workspace);
+
+    const onShareDialogBaseError = useCallback(
+        (err: Error) => {
+            onError && onError(new UnexpectedSdkError(err.message, err));
```

Comment:
Nitpick: `onError?.(new UnexpectedSdkError(err.message, err));` would be cleaner.

Reply:
what ??? this is possible ? :-)

Reply:
Done